src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -268,34 +274,39 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 // exception from the exception.)
                 // This is the dual to the special exception for offset-by-0
                 // in the inbounds pointer offset operation (see the Miri code, `src/operator.rs`).
-                if a.is_bits() && b.is_bits() {
+                //
+                // Control flow is weird because we cannot early-return (to reach the
```

Comment:
we have a bunch of situations like that in miri... Maybe we should consider using `try` blocks at some point